src/lib/signature_kind/type/mina_signature_kind_type.ml

```diff
@@ -1 +1,4 @@
+open Core_kernel
+
 type t = Testnet | Mainnet | Other_network of string
+[@@deriving bin_io_unversioned]
```

Comment:
This `[@@deriving bin_io_unversioned]` and the `[@@deriving bin_io]` in the `mina_signature_kind.mli` above should be the right way of doing this. It's what I had to do [in my original draft for this change](https://github.com/MinaProtocol/mina/pull/16976/files#diff-3a7eff8a40cf4cb05c1797d8b4276ddbd40a3b756754d5d4792336c8a960059a).

Reply:
If we ever wanted to use the signature kind externally, we'd need to put it in a versioned module. But at that point we could just change from `bin_io_unversioned` to `bin_io`.